### PR TITLE
Fixes #3572 and some others: Remove analyzers from solution

### LINKED ISSUE #3450: Can't build the current v2_develop branch 37f59d2
I get erros like this:

`CS0016	Could not write to output file '..\gui-cs\Terminal.Gui\Analyzers\Terminal.Gui.Analyzers.Internal\obj\Debug\netstandard2.0\generated\Meziantou.Polyfill\Meziantou.Polyfill.PolyfillGenerator\M_System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd``1(`0,System.Func{`0,``0,`1},``0).g.cs' -- 'Could not find a part of the path '..\gui-cs\Terminal.Gui\Analyzers\Terminal.Gui.Analyzers.Internal\obj\Debug\netstandard2.0\generated\Meziantou.Polyfill\Meziantou.Polyfill.PolyfillGenerator\M_System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd``1(`0,System.Func{`0,``0,`1},``0).g.cs'.'	Terminal.Gui.Analyzers.Internal	..\gui-cs\Terminal.Gui\Analyzers\Terminal.Gui.Analyzers.Internal\CSC
`

The main README file doesn't mention anything about what this might happen if someone wants to clone the repository.

### LINKED ISSUE #3466: Analyzers project builds with warnings
@dodexahedron can you please fix all the warnings?

```
4>------ Build started: Project: Terminal.Gui, Configuration: Debug Any CPU ------
2>C:\Program Files\dotnet\sdk\8.0.300-preview.24203.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(311,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy
3>C:\Users\Tig\s\gui-cs\Terminal.Gui\Analyzers\Terminal.Gui.Analyzers.Internal.Tests\Generators\EnumExtensions\EnumDefinitions\WithGenerator\BetterEnum_ExplicitInt_NoFastIsDefined.cs(44,5,44,10): warning CA1707: Remove the underscores from member name Terminal.Gui.Analyzers.Internal.Tests.Generators.EnumExtensions.EnumDefinitions.BetterEnum_ExplicitInt_NoFastIsDefined.All_0 (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1707)
3>C:\Users\Tig\s\gui-cs\Terminal.Gui\Analyzers\Terminal.Gui.Analyzers.Internal.Tests\Generators\EnumExtensions\EnumDefinitions\WithGenerator\BetterEnum_ExplicitInt_NoFastIsDefined.cs(46,5,46,19): warning CA1707: Remove the underscores from member name Terminal.Gui.Analyzers.Internal.Tests.Generators.EnumExtensions.EnumDefinitions.BetterEnum_ExplicitInt_NoFastIsDefined.Alternating_01 (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1707)
3>C:\Users\Tig\s\gui-cs\Terminal.Gui\Analyzers\Terminal.Gui.Analyzers.Internal.Tests\Generators\EnumExtensions\EnumDefinitions\WithGenerator\BetterEnum_ExplicitInt_NoFastIsDefined.cs(45,5,45,10): warning CA1707: Remove the underscores from member name Terminal.Gui.Analyzers.Internal.Tests.Generators.EnumExtensions.EnumDefinitions.BetterEnum_ExplicitInt_NoFastIsDefined.All_1 (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1707)
3>C:\Users\Tig\s\gui-cs\Terminal.Gui\Analyzers\Terminal.Gui.Analyzers.Internal.Tests\Generators\EnumExtensions\EnumDefinitions\WithGenerator\BetterEnum_ExplicitInt_NoFastIsDefined.cs(47,5,47,19): warning CA1707: Remove the underscores from member name Terminal.Gui.Analyzers.Internal.Tests.Generators.EnumExtensions.EnumDefinitions.BetterEnum_ExplicitInt_NoFastIsDefined.Alternating_10 (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1707)
3>C:\Users\Tig\s\gui-cs\Terminal.Gui\Analyzers\Terminal.Gui.Analyzers.Internal.Tests\Generators\EnumExtensions\EnumDefinitions\WithGenerator\BetterEnum_NoFastIsDefined.cs(43,5,43,10): warning CA1707: Remove the underscores from member name Terminal.Gui.Analyzers.Internal.Tests.Generators.EnumExtensions.EnumDefinitions.BetterEnum_NoFastIsDefined.All_0 (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1707)
3>C:\Users\Tig\s\gui-cs\Terminal.Gui\Analyzers\Terminal.Gui.Analyzers.Internal.Tests\Generators\EnumExtensions\EnumDefinitions\WithGenerator\BetterEnum_NoFastIsDefined.cs(44,5,44,10): warning CA1707: Remove the underscores from member name Terminal.Gui.Analyzers.Internal.Tests.Generators.EnumExtensions.EnumDefinitions.BetterEnum_NoFastIsDefined.All_1 (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1707)
3>C:\Users\Tig\s\gui-cs\Termin
...
```

### LINKED ISSUE #3477: `Analyzers` -> Restarting VS when simply changing branches is not acceptable.
Every time I switch branches I need to exit VS, re-build them, and restart VS. This is not acceptable.

On my fast desktop this is painful. On my laptop it's excruciating. It takes many minutes just to change branches and I do that frequently.  

@dodexahedron can you please find a solution to this that does not require such a painful workflow. When I agreed to merge it, I was under the impression that I'd only have to do the song-and-dance if I actually changed something that utilized them. But that's not the case, apparently.

Given the current value `Analyzers` provides is extremely limited I'm prone to just ask you to yank it out for now unless there's a way of them working that doesn't provide such a cruddy developer experience.

### LINKED ISSUE #3494: Simplify workflow surrounding roslyn projects
Quick Version:

Going to split the Roslyn-related projects off to their own solution, pack as a nuget package, and reference that in Terminal.Gui.csproj.

### The Problem
Initially, I was a bit over-optimistic about the impact to some workflows these would have, since I hadn't considered the fact that other project members still end up switching branches between v1 and v2 as needed for bugfixes.

Unfortunately, Visual Studio just doesn't make things easy to begin with, but everything involved in switching back and forth just compounds the issue for everyone.

### Available Workarounds
There are some available workarounds to ease those workflows, including:
 - There's a VS extension that takes advantage of how assembly loading works and just gives roslyn projects a unique name on every build, so a brand now assembly can be loaded.
 - Contributors could clone v1 and v2 to two separate work trees so switching between v1 and v2 isn't required.

But these are both non-ideal to force people to do, and I certainly don't want to depend on that stuff either.

So....

### Solution
Pun of course intended.

The solution that requires the least interaction and fewest workflow changes for team members is to just go all the way and pack any Roslyn generator projects as NuGet packages, and then to reference those packages like any other in the projects that need them.

### Options for nuget packaging
There are a few ways to go about this, but at a basic level it comes down to 2 general options:
 1. Pack into a local nuget repository and consume the packages from there.
    - This is the simplest on the initial setup of things and can be done entirely within branches here.
    - This does, however, have drawbacks in that it's placing the responsibility of packing the nupkg on any contributor and it still needs to be done before the package can be successfully restored. Pre-build events can handle that sort of thing, but it can potentially mean slowing local builds down a bit, plus spurious errors being flagged by VS until restore happens, which is kinda janky if you ask me.
 2. Pack into a nupkg for consumption via a remote repository.
    a. I was already planning to make the minor changes necessary to make the existing generator project-agnostic and publish on NuGet from the SnapsInAZfs organization, so it could just live there.
    b. We could also make a new repo in the gui-cs organization, if desired.
    c. For either of those, the code will be the same. 

None of these are mutually exclusive - just redundant if more than one is done.

### The Plan
I intend to do 2a no matter what, especially since it doesn't innately have anything to do with TG other than a few strings, and is quite likely to get continued development from my company in the future that would just be noise to have over here.

In any case, the work to do this is relatively minimal.

The basic/rough procedure is:

 1. Remove all 3 of the Roslyn projects that live in the Analyzers directory from Terminal.sln and remove corresponding ProjectReference elements for them in any projects that have them in Terminal.sln.
 2. Clone the Analyzers tree to a repository specifically for them.
 3. Set up NuGet packaging for it and publish.
 4. In Terminal.Gui.csproj and anywhere else that needs it, reference that package like any other.
 5. Live happily ever after.

That looks to me like probably 2 or 3 commits on the TG side.

The part that will take the longest is just administrative junk, since only superficial code changes are needed.

The Terminal.Gui.PowerShell.Analyzers powershell module will be rendered obsolete by this, too, so I'll delete that at the end, once it's all set up.

### LINKED ISSUE #3544: v2 crashes on run with missing dependency JetBrains.Annotations
**Describe the bug**
When referencing the latest v2 Terminal.Gui nuget package, starting up crashes:

```
System.IO.FileNotFoundException: 'Could not load file or assembly 'JetBrains.Annotations, Version=4242.42.42.42, Culture=neutral, PublicKeyToken=1010a0d8d6380325'. The system cannot find the file specified.'
```

**To Reproduce**
Create a new project:
```
dotnet new console -n "repro"
dotnet add package Terminal.Gui --prerelease
```

Set Program.cs to:

```
using Terminal.Gui;

Application.Init();

Application.Run(new Window());

Application.Shutdown();
```

**Expected behavior**
Program should start up, instead you get the above exception

As a workaround you can manually add the JetBrains.Annotations dependency into your project.

### LINKED ISSUE #3572: Some more prep for revamped sourcegens and CI
This issue is primarily for the next piece of preparation for a couple of other issues, such as splitting off the enum generators and enhancements to the CI pipeline that kinda come along for free with that.

But it's also a good spot to bring in a couple other related but small pieces of work which also share the goal of simplification of the build and also the consumer's experience with the packaged library.

At a high level, this one will come with at least:

 * [x] Additional configurations added to the solution and projects for which they are relevant.
    * At minimum, some specifically meant for builds/tests/deployments in CI, which also aim to improve the design-time experience by making that stuff not part of local builds unless you want it to be, by choosing the CI configurations.
 * [ ] Take a pass over dependencies and reduce where feasible.
    * Highest priority there being reducing **run-time** dependencies on code consuming TG (such as the root of #3544).
 * [ ] Include a pre-packaged form of the enum generators and remove the projects from the solution.
    * [ ] Also going to drop a README.md in the Analyzers and project directories under Analyzers with a note that they're deprecated and slated for replacement soon, just in case someone comes along and wants to add onto them or something. 🤷‍♂️ 
 * [ ] Declare existing classes in the Roslyn projects as `partial` and add a conditionally-compiled file that adds `[Obsolete]` to all of them, with the condition not being true by default until later, by flipping a switch, when the change to using the external package is almost ready to go.
    * I'll make a PR for that switch flip when it's time.
    * This is really just an insurance policy, but it's almost 0-effort, so might as well go for it.
 * Probably other minor stuff I'll mention if/when they come up.

### PR DESCRIPTION
**Pull request by dodexahedron**
This PR removes the enum source generator projects, code, pwsh module, and CI build item from v2.

The generated code from the v2 code at the base of this branch was copied into normal source files so that existing uses can be preserved.

Note that, if you change one of those enums, you need to add an entry to the FastIsDefined method's switch expression for that enum's extension class. Enum changes are considered breaking changes in the .net world, anyway, however.

Those files will get deleted at whatever point in the future I come back to the generators. Probably during beta.

## Fixes

- Fixes #3572
- Fixes #3494 
- Fixed #3544 in my environment, but not an explicit goal of this PR
- Fixes #3477 
- Fixes #3466 
- Fixes #3450 
- Postpones #3443 

## Proposed Changes/Todos

- [x] Remove analyzer projects
- [x] Remove analyzers from CI build
- [x] Remove Terminal.Gui.PowerShell.Analyzers PowerShell module
- [x] Preserve currently used generated code
- [x] Other miscellaneous cleanup mandated by the above

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [ ] ~~I have made corresponding changes to the API documentation (using `///` style comments)~~
       - Not relevant 
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

Ready to go

### CODE REVIEW

**Review by BDisp — APPROVED** [review]
LGTM. I switch from develop and this PR branches and rebuild on both and it is working as expected without needing to run any scripts. Thanks.

**Comment by dodexahedron** [comment]
Good to hear.

I had tried in a couple of environments as well, but still wanted to be extra-sure I wasn't just getting lucky. 😅 

I was originally splitting them off into a local nuget package but decided it wasn't worth wasting the effort on that since I'm _already_ going to be doing something similar but not identical when I get back to it in the formal public nuget package I'll be putting out eventually.

Just wanted to work on other more important stuff for now and make your workflows easier in as simple a way as possible. There are other ways to alter your workflow that would have avoided the pain you guys were feeling, but it didn't feel worth altering those habits for something temporary.

**Review by tig — COMMENTED** [review]
LGTM!

Thank you. Thank you. Thank you.

**Comment by BDisp** [comment]
@dodexahedron It looks like you removed the unit tests. It would be beneficial to create them in the `UnitTests` project, but using `TestHelpers` to normalize the newlines regardless of whether the `WSL` tests are using the `Windows 11` file system or not.

File: Terminal.Gui/Terminal.Gui.csproj [hunk 1]
```diff
@@ -59,11 +59,6 @@
     <PackageReference Include="System.IO.Abstractions" Version="21.0.22" />
     <PackageReference Include="System.Text.Json" Version="8.0.4" />
     <PackageReference Include="Wcwidth" Version="2.0.0" />
-    <ProjectReference Include="..\Analyzers\Terminal.Gui.Analyzers.Internal\Terminal.Gui.Analyzers.Internal.csproj">
-      <PrivateAssets>all</PrivateAssets>
-      <OutputItemType>Analyzer</OutputItemType>
-      <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
-    </ProjectReference>
```
Comment (BDisp) [inline Terminal.Gui/Terminal.Gui.csproj:66]: You removed it here in #3600 but it's again on the` v2_develop`.

https://github.com/gui-cs/Terminal.Gui/blob/7dc12a4c5844c1446df6ff2b337481ac89e5e609/Terminal.Gui/Terminal.Gui.csproj#L62-L66

Comment (dodexahedron) [reply Terminal.Gui/Terminal.Gui.csproj:66]: I removed the analyzer project and its tests, yes.

Didn't remove the actual TG UnitTests project.

But good thought on reformatting and whatnot. I'll add what's necessary to make ReSharper ignore those files as part of #3558 too, so it doesn't change things.

**Pull request merged**